File: src/main/java/com/example/progettoenterprise/dto/ImmagineViaggioDTO.java
package com.example.progettoenterprise.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ImmagineViaggioDTO {
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private Long id;

    @NotBlank(message = "L'URL dell'immagine è obbligatorio")
    @Size(max=500, message = "Limite di caratteri del link superato")
    private String url;

    @NotNull(message = "Specificare se l'immagine deve essere pubblica o privata")
    private boolean pubblica;

    private Long viaggioId;

}

File: frontendMobile/app/src/main/java/com/example/enterprisemobile/model/dto/ImmagineViaggioDTO.kt
package com.example.enterprisemobile.model

import com.google.gson.annotations.SerializedName

data class ImmagineViaggioDTO(
    @SerializedName("id") val id: Long,
    @SerializedName("url") val url: String,
    @SerializedName("pubblica") val pubblica: Boolean,
    @SerializedName("viaggioId") val viaggioId: Long
)